src/OpenToolkit.Mathematics/Geometry/Box2.cs

```diff
@@ -121,6 +121,160 @@ public Vector2 Center
             set => Translate(value - Center);
         }
 
+        // --
+
+        public float Width
+        {
+            get => _max.X - _min.X;
+            set => _max.X = _min.X + value;
+        }
+
+        public float Height
+        {
+            get => _max.Y - _min.Y;
+            set => _max.Y = _min.Y + value;
+        }
+
+        public float Left
+        {
+            get => _min.X;
+            set => _min.X = value;
+        }
+
+        public float Top
+        {
+            get => _min.Y;
+            set => _min.Y = value;
+        }
+
+        public float Right
+        {
+            get => _max.X;
+            set => _max.X = value;
+        }
+
+        public float Bottom
+        {
+            get => _max.Y;
+            set => _max.Y = value;
+        }
+
+        public float X
+        {
+            get => _min.X;
+            set => _min.X = value;
+        }
+
+        public float Y
+        {
+            get => _min.Y;
+            set => _min.Y = value;
+        }
+
+        public float SizeX
+        {
+            get => _max.X - _min.X;
+        }
+
+        public float SizeY
+        {
+            get => _max.Y - _min.Y;
+        }
+
+        public Vector2 Location => _min;
+
+        public bool IsEmpty => _min.X == 0 && _min.Y == 0 && _max.X == 0 && _max.Y == 0;
```

Comment:
I think we should consider a different name for this, because I think IsEmpty can be confusing if you are doing stuff like collision-detection, and the name doesn't really say what it returns (a box with nothing in it could also be considered empty).

I think we could also add a `IsDegenerate` property to check if one of the sides have zero length.

Reply:
What about IsZero?

Reply:
I would be fine with that I guess. Another name could be `HasZeroSize` or something that says it's size is zero, `IsAreaZero` could also be an idea.